app/components/app/app.service.js

```diff
@@ -1,29 +1,64 @@
 class DataService {
-    constructor($http, BASE_URL) {
+    constructor(firebase, toastService, $firebaseObject) {
         'ngInject';
 
-        this.$http = $http;
-        this.BASE_URL = BASE_URL;
+        this.firebase = firebase;
+        this.toastService = toastService;
+        this.$firebaseObject = $firebaseObject;
     }
 
     getCategories() {
-        return this.$http.get(`${this.BASE_URL}/categories.json`)
-            .then((response) => {
-                return response.data;
-            })
-            .catch(this.responseError);
+        const dbRefObjectCategories = this.firebase.database().ref('api/v1').child('categories');
+        const categoriesData = this.$firebaseObject(dbRefObjectCategories);
+
+        return categoriesData.$loaded().then(responseCategories => {
+            return responseCategories;
+        });
+    }
+
+    getCategory(categoryKey) {
+        const dbRefObjectCategory = this.firebase.database().ref('api/v1/categories').child(categoryKey);
+        const categoryData = this.$firebaseObject(dbRefObjectCategory);
+
+        return categoryData.$loaded().then(responseCategory => {
+            return responseCategory;
+        });
+    }
+
+    getProductsByCategory(category) {
+        const dbRefObjectProducts = this.firebase.database().ref('api/v1').child('products').orderByChild('category').equalTo(category);
```

Comment:
In case of so long lines you can split it into new lines, like:

```js
const dbRefObjectProducts = 
    this.firebase.database()
        .ref('api/v1')
        .child('products')
        .orderByChild('category')
        .equalTo(category);
```